### DISCUSSION TITLE
random movement of servo connected to GPIO 34

### DISCUSSION THREAD

**Discussion by spikey1973** [micropython/micropython#10382] [ESP32] [answered]
hello to all here, 

i am hobbying here on my holiday, and have managed to get many steps further over these days thanks to all the help here. 
Unfortunately now i am struggling with an issue that i don't know how to start with. 

I have two servo's (20kg) connected to my wemos esp32 s2 mini board. the servo's are of the same make.. 

1) is connected to pin / GPIO6, and it's doing random movement. 
both random in timing, and the movement itself (direction, and degrees), mostly at start up, but not just. 

2) is connected to pin / GPIO34, and is does not do this.. 

besides that i have also connected, 2x vibration sensor sw-240, 2x drv8825 stepper driver, 2x  TCST2103 optical end switch and a ttp223 touch sensor connected to that board. 

besides the boot.py, i have no code running, nor have had ever installed, just used the repl through data cable. 

I posted the boot.py code here below, just in case, but i do not feel there could be an issue there.. 

Never the less there is a cause..
as far as i know I thought of three possible causes.
1) wrong pin connection, with this i mean it is connected to a pin that is not ideal for servo connections and has another PWM function for which it send these pulses. 
2) short / interferense in my build (i have not been able to connect this) 
3) an internal issue of the servo itself. 

point 2 and 3 is something i need to deal with myself ofcourse, but if there is an issue concerning point 1 i do not feel i have enough knowledge to decide a yes or no on this. 

pin specs according to the esp32 s2 mini datasheet. 
GPIO6: IO6  I/O/T RTC_GPIO6, GPIO6, TOUCH6, ADC1_CH5 
GPIO34: IO34 I/O/T SPIIO5, GPIO34, FSPICS0

Has anyone any thoughts on this? 

Kind great, Matthieu 

the boot.py code 

```
def do_connect():
    import network
    wlan = network.WLAN(network.STA_IF)
    wlan.active(True)
    if not wlan.isconnected():
        print('connecting to network...')
        wlan.connect('DrayTek', '*********')
        while not wlan.isconnected():
            pass
    print('network config:', wlan.ifconfig())

do_connect()

# This file is executed on every boot (including wake-boot from deepsleep)
#import esp
#esp.osdebug(None)

from machine import Pin
from neopixel import NeoPixel
pin_1_P_LLC1 = Pin(1, Pin.OUT)
leds = NeoPixel(pin_1_P_LLC1, 8)
leds[0] = (0, 0, 0)  # Off
leds.write()

import webrepl
webrepl.start()
```

**Comment by davefes** [comment]
Some pins are used for different purposes during boot.  Search "strapping pins".  I can tell you which ones to avoid on the ESP32-WROOM but they might not apply to your variant.

**Reply by spikey1973** [reply]
For the esp s2 mini the strapping pins are not used on the board. in this case the GPIO0,45,46. Never the less, i was not aware of this fact so it has been an informative responce. Thank you!

**Comment by peterhinch** [comment, marked as answer]
On boot all pins are high impedance. This may cause unpredictable behaviour. You need to configure as outputs the pins connected to the servos.

**Reply by spikey1973** [reply]
Hello Peter, 

Yeah I found that out. As I was Just dealing with the Boot.py I had not yet configured the pins yet. 
Unfortunately., by the time I realised that a short happened because of all m fiddling and now part of the board is no longer responding to in/outputs. SO I am kinda annoyed with myself.
